cmd/chainbridge/account.go

```diff
@@ -85,12 +75,36 @@ func handleImportCmd(ctx *cli.Context, dHandler *dataHandler) error {
 	// import key
 	if keyimport := ctx.Args().First(); keyimport != "" {
 		var err error
+
 		log.Info("Importing key...")
+
+		// check if --ed25519 or --sr25519 is set
+		keytype := crypto.Secp256k1Type
+		if flagtype := ctx.Bool(Sr25519Flag.Name); flagtype {
+			keytype = crypto.Sr25519Type
+		} else if flagtype := ctx.Bool(Secp256k1Flag.Name); flagtype {
+			keytype = crypto.Secp256k1Type
+		}
+
 		if ctx.Bool(EthereumImportFlag.Name) {
 			_, err = importEthKey(keyimport, dHandler.datadir)
+		} else if privkeyflag := ctx.String(PrivateKeyFlag.Name); privkeyflag != "" {
+			// check if --password is set
+			var password []byte = nil
+			if pwdflag := ctx.String(PasswordFlag.Name); pwdflag != "" {
+				password = []byte(pwdflag)
+			}
+
+			// Hex must not have leading 0x
+			if privkeyflag[0:2] == "0x" {
+				_, err = importPrivKey(keytype, privkeyflag[0:2], dHandler.datadir, password)
```

Comment:
You've mixed up the parameters: `datadir` is the second and `key` is the third, and `privkeyflag[0:2]` is not parsing the private key correctly

Reply:
all instances of `importPrivKey` seem to be mkaing the same error of mixing params